### PR TITLE
Request images without team selected

### PR DESCRIPTION
**Pull request by henleo**

### CODE REVIEW

File: src/DialogflowEntityClient.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+export class EntityNotFoundError extends Error { }
+
+export class DialogflowEntityClient {
+  constructor(entityTypeName, projectId) {
+    this.entityTypeName = entityTypeName;
+    this.projectId = projectId;
+  }
+
+  async initialize() {
```
Comment (arne-z) [inline src/DialogflowEntityClient.js]: Could use lazy initialization with [this](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Functions/get).

File: src/controller/DialogflowController.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
 // @flow
 import type { $Response as Response, $Request as Request } from 'express';
-import { EntityTypesClient } from 'dialogflow';
+// import { EntityTypesClient } from 'dialogflow';
```
Comment (arne-z) [inline src/controller/DialogflowController.js]: Nopls.

File: src/DialogflowEntityClient.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    };
+    return request;
+  }
+
+  async createEntity(value, synonyms) {
+    await this.initialize();
+    const entityType = await this.entityTypeObject();
+    const newTeam = { synonyms, value };
+    entityType.entities.push(newTeam);
+    const request = DialogflowEntityClient.createRequest(entityType);
```
Comment (Adrian-St) [inline src/DialogflowEntityClient.js]: I prefer this.constructor.createRequest(entityType)

File: src/controller/DialogflowController.js [hunk 4]
```diff
@@ -104,50 +107,25 @@ export function validateTeamsParams(req: Request, res: Response, next: any) {
...
+    return res.sendStatus(200);
+  } catch (err) {
+    if (err instanceof EntityNotFoundError) {
+      logger.info('Could not find the entity named Team.');
+      return res.status(500).send('Could not find entity type Dialogflow.');
+    }
+    logger.info(`Error updating entity type: ${err}`);
+    return res.status(500).send('Could not update Team on Dialogflow.');
+  }
 }
```
Comment (Adrian-St) [inline src/controller/DialogflowController.js]: Nice!

File: src/controller/DialogflowController.js (hunk 2 above)
Comment (Adrian-St) [inline src/controller/DialogflowController.js]: Remove pls

File: src/DialogflowEntityClient.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  static createRequest(entityTypeObject) {
+    const request = {
+      entityType: entityTypeObject,
+      updateMask: {
+        paths: ['entities'],
+      },
+    };
+    return request;
```
Comment (Adrian-St) [inline src/DialogflowEntityClient.js]: return object directly instead

File: src/DialogflowEntityClient.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  async entityTypeObject() {
+    const entityTypesList = await this.entityTypesList();
+    const entity = entityTypesList[0]
+      .find(resource => resource.displayName === this.entityTypeName);
+    if (entity) {
+      return entity;
+    }
+    throw EntityNotFoundError;
```
Comment (Adrian-St) [inline src/DialogflowEntityClient.js]: Please do if(!entity) { throw EntityNotFoundError }.
Side effects should not be the last statement.

**Pull request merged**